File: projecteuler/PE_002/Julia/PE_002_naive.jl
function PE_002_naive(limit=4 * 10^6, F_1=1, F_2=2)
    total = 0
    while F_2 < limit
        if F_2 % 2 == 0
            total += F_2
        end
        F_1, F_2 = F_2, F_1 + F_2
    end
    total
end

File: projecteuler/PE_002/Python/PE_002_naive.py
def fibonacci(limit=float("Inf")):
    fib, fib_next = 0, 1
    while fib < limit:
        yield fib
        fib, fib_next = fib_next, fib + fib_next


def PE_002_naive(limit=4 * 10 ** 6):
    total = 0
    for fib in fibonacci(limit):
        if fib % 2 == 0:
            total += fib
    return total


def _PE_002_naive_functional(limit=4 * 10 ** 6):
    return sum(fib for fib in fibonacci(limit) if fib % 2 == 0)


if __name__ == "__main__":

    print(PE_002_naive())
